### PR TITLE
update cdns to cloudflare

### PR DESCRIPTION
**Pull request by illcat**
#423 Add sha512-code and crossorigin.

### CODE REVIEW

File: _includes/custom-head.html [hunk 1]
```diff
@@ -7,18 +7,18 @@
...
-    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"> </script>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css" integrity="sha512-h7nl+xz8wgDlNM4NqKEM4F1NkIRS17M9+uJwIGwuo8vGqIl4BhuCKdxjWEINm+xyrUjNCnK5dCrhM0sj+wTIXw==" crossorigin="anonymous" />
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.js" integrity="sha512-/CMIhXiDA3m2c9kzRyd97MTb3MC6OVnx4TElQ7fkkoRghwDf6gi41gaT1PwF270W6+J60uTmwgeRpNpJdRV6sg==" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/contrib/auto-render.min.js" integrity="sha512-Do7uJAaHZm5OLrIv/yN4w0iG1dbu01kzdMNnFfu/mAqgUk6Nniv2JYHcwH+cNwjqgLcqcuBBk+JRvprLVI8azg==" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.min.js" integrity="sha512-XwnBT4t1/cl+Qz3T187hcrLT+TQ/HJ58wGk3Tj+uZuundcWdri5cfQV/NIPbM6gtfyVznybMBU1R2IPShJfEdA==" crossorigin="anonymous"></script>
```
Comment (hamelsmu) [inline _includes/custom-head.html:21]: For mathjax, these two sites give me different javascript, which is a bit suspect for me

"https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.min.js" "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.min.js"

Comment (illcat) [reply _includes/custom-head.html]: fix

**Comment by hamelsmu** [comment]
Can you not delete `_action_files/check_js.sh` and `.github/workflows/check_cdns.yaml` - I would still like to keep these files

**Comment by illcat** [comment]
> Can you not delete `_action_files/check_js.sh` and `.github/workflows/check_cdns.yaml` - I would still like to keep these files

integrity="sha512-******************" can achieve the function, and it can prevent the time difference of CDN synchronization.

**Comment by hamelsmu** [comment]
Totally understand, I would like to keep these files- I will deactivate the workflow but still keep the files

**Comment by hamelsmu** [comment]
I'll just merge and add back in myself

**Pull request merged**